bitnami/seaweedfs external postgres database support

wkbrd opened the pull request:
### Description of the change

Add support for external database for seaweedfs to be provided by a PostgreSQL database.

### Benefits

Enables PostgreSQL as an external database choice in addition to existing MariaDB option.

### Possible drawbacks

N/A

### Applicable issues

- implements #29215
- builds off work in #29196 

### Additional information

Resulting PR implements both #29215 and #29196.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

juan131 commented on bitnami/seaweedfs/README.md:
```diff
@@ -1043,7 +1045,8 @@ enabled = false
 | `mariadb.primary.resources`                | Set container requests and limits for different resources like CPU or memory (essential for production workloads)                                                                                                          | `{}`                |
-| `externalDatabase.enabled`                 | Enable secret with external mariadb database credentials                                                                                                                                                                   | `false`             |
+| `externalDatabase.enabled`                 | Enable external database database support                                                                                                                                                                                  | `false`             |
+| `externalDatabase.dbms`                    | Database management system product (mariadb, postgresql)                                                                                                                                                                   | `mariadb`           |
```
I don't know if `dbms` is the most appropriate name for the new parameter. Maybe "store" to be aligned with upstream terms? See:

- https://github.com/seaweedfs/seaweedfs/wiki/Filer-Stores

wkbrd replied:
Will update to 'store'

juan131 reviewed (changes requested):
Thanks so much for this awesome contribution! Please check my comments when you have a chance.

wkbrd commented on bitnami/seaweedfs/templates/externaldb-secrets.yaml:
```diff
@@ -15,5 +15,5 @@ metadata:
 type: Opaque
 data:
-  mariadb-password: {{ .Values.externalDatabase.password | b64enc | quote }}
+  {{ .Values.externalDatabase.dbms }}-password: {{ .Values.externalDatabase.password | b64enc | quote }}
```
Definitely OK with this recommendation.

wkbrd commented:
> Thanks so much for this awesome contribution! Please check my comments when you have a chance.

@juan131 , I think I have addressed all of your feedback. Please review when you have a moment. Thanks!

wkbrd commented:
@juan131 : The build seems to be timing out on VIB Verify.  Anyone know how to correct that or what that means?

wkbrd commented:
@juan131 : It is now passing. I had to make it have mariadb-password for the mariadb.enabled since the secret key name in bitnami/mariadb is named that way.

juan131 commented on bitnami/seaweedfs/templates/_helpers.tpl:
```diff
@@ -283,6 +283,19 @@ Return the database secret name
+{{- define "seaweedfs.database.keyName" -}}
+{{- if .Values.mariadb.enabled }}
+    {{- print "mariadb-password" -}}
+{{- else if .Values.externalDatabase.existingSecret -}}
+    {{- printf "%s-password" .Values.externalDatabase.scope -}}
```
Typo:
```suggestion
    {{- printf "%s-password" .Values.externalDatabase.store -}}
```

wkbrd replied:
Addressed. Good find!

juan131 commented on bitnami/seaweedfs/templates/_helpers.tpl:
```diff
@@ -283,6 +283,19 @@ Return the database secret name
+{{- else if .Values.externalDatabase.existingSecret -}}
+    {{- printf "%s-password" .Values.externalDatabase.scope -}}
+{{- else -}}
+    {{- printf "password" -}}
```
No need for `printf`:
```suggestion
    {{- print "password" -}}
```

juan131 reviewed (changes requested):
@wkbrd in order to use PostgreSQL, users should ensure they previously create the `filemeta` table with the corresponding schema. Sth like this:

```sql
\c bitnami_seaweedfs;
CREATE TABLE IF NOT EXISTS filemeta (
  dirhash     BIGINT,
  name        VARCHAR(65535),
  directory   VARCHAR(65535),
  meta        bytea,
  PRIMARY KEY (dirhash, name)
);
```

When `mariadb.enabled` is `true`, this is done using the `mariadb.initdbScripts` parameter. However, when using an externally managed PostgreSQL it might be worth to throw some warning in the NOTES (or similar) to alert users.

wkbrd commented:
> @wkbrd in order to use PostgreSQL, users should ensure they previously create the `filemeta` table with the corresponding schema. Sth like this:
> 
> ```sql
> \c bitnami_seaweedfs;
> CREATE TABLE IF NOT EXISTS filemeta (
>   dirhash     BIGINT,
>   name        VARCHAR(65535),
>   directory   VARCHAR(65535),
>   meta        bytea,
>   PRIMARY KEY (dirhash, name)
> );
> ```
> 
> When `mariadb.enabled` is `true`, this is done using the `mariadb.initdbScripts` parameter. However, when using an externally managed PostgreSQL it might be worth to throw some warning in the NOTES (or similar) to alert users.

I have updated the README.md to describe this requirement.  This change improves the documentation for both mariadb and postgresql.  Let me know if further changes are needed.

wkbrd commented:
@juan131 : I have made updates based on all of your feedback. Please re-review when you have a moment. Thanks!

juan131 reviewed (approved):
Awesome work! LGTM
